app/src/main/java/com/example/dashcarr/presentation/tabs/map/SessionInformationDrawable.kt

```diff
@@ -0,0 +1,154 @@
+package com.example.dashcarr.presentation.tabs.map
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.ColorFilter
+import android.graphics.Paint
+import android.graphics.PixelFormat
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
+import android.location.Location
+import com.android.volley.Request
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.Volley
+import kotlin.math.roundToInt
+
+
+class SessionInformationDrawable(
+    private val context: Context,
+    private var textSizes: Array<Float>,
+    private val rotateDrawing: Boolean,
+    private var onSizeChanged: (Int) -> Unit
+) : Drawable() {
+    private val paint: Paint = Paint()
+    private var street: String = ""
+    private lateinit var location: Location
+    private var missingStreetUpdate = true
+
+    fun updateLocation(location: Location) {
+        if (missingStreetUpdate || this.location.distanceTo(location) > 50) {
+            updateStreet(location)
+        }
+        this.location = location
+        this.invalidateSelf()
+    }
+
+    private fun updateStreet(location: Location) {
```

Comment:
testing on my avd don't get the address in both mapview and hud

Reply:
The reason for that is probably that your IP address is blocked due to too many requests, because of fast movement. I changed the code to reduce the number of requests. 
Try this Url: https://nominatim.openstreetmap.org/reverse?format=jsonv2&lat=68.43671166666667&lon=17.426035&layer=address
If it works, it should work on the app as well. If you are still blocked a VPN is solving it temporarily. I think they let you free after 24h.